apple/swift-docc-render
# Use spinner icon to indicate indefinite progress while loading navigator

Bug/issue #, if applicable: 92438279

## Summary

This updates the navigator UI to use a spinner icon to indicate indefinite loading progress instead of the **"Fetching..."** text. \cc @biamx3

![spinner](https://user-images.githubusercontent.com/212918/166331597-88c23be0-ee9a-4dcf-a685-a4c687c47008.gif)

## Testing

You can use the following patch to simulate a scenario where the navigator is taking infinitely long to fetch index data:

```diff
diff --git a/src/views/DocumentationTopic.vue b/src/views/DocumentationTopic.vue
index 4770b61..c22020b 100644
--- a/src/views/DocumentationTopic.vue
+++ b/src/views/DocumentationTopic.vue
@@ -42,7 +42,7 @@
                 <Navigator
                   :parent-topic-identifiers="parentTopicIdentifiers"
                   :technology="slotProps.technology || technology"
-                  :is-fetching="slotProps.isFetching"
+                  :is-fetching="true"
                   :error-fetching="slotProps.errorFetching"
                   :api-changes="slotProps.apiChanges"
                   :references="topicProps.references"

```

Steps:
1. Copy the patch above and apply it with `pbpaste | git apply`
2. Run the dev server with example data using `env VUE_APP_DEV_SERVER_PROXY=https://ethan-kusters.github.io/swift-argument-parser npm run serve`
3. Open an example URL like http://localhost:8080/documentation/argumentparser and verify that the navigator shows the spinner icon instead of the text **Fetching...**

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
